File: tests/test-minecraft/src/main/java/pize/tests/minecraft/run/Minecraft.java
package pize.tests.minecraft.run;

import pize.Pize;
import pize.app.AppAdapter;
import pize.graphics.gl.DepthFunc;
import pize.graphics.gl.Gl;
import pize.graphics.gl.Target;
import pize.tests.minecraft.game.Session;
import pize.tests.minecraft.utils.log.Logger;

public class Minecraft extends AppAdapter{

    public static void main(String[] args){
        Pize.create("Minecraft", 925, 640);
        Pize.window().setIcon("icon.png");
        Pize.run(new Minecraft());
    }


    private Session session;

    @Override
    public void init(){
        Gl.enable(Target.DEPTH_TEST);
        Gl.depthFunc(DepthFunc.LEQUAL);
        Thread.currentThread().setName("Render Thread");

        session = new Session();
    }

    @Override
    public void render(){
        Gl.clearColorDepthBuffers();
        session.render();
    }

    @Override
    public void resize(int width, int height){
        session.resize(width, height);
    }

    @Override
    public void dispose(){
        session.dispose();

        Logger.instance().info("EXIT.");
    }

}

File: tests/test-minecraft/src/main/kotlin/pize/tests/minecraft/run/Minecraft.kt
package pize.tests.minecraft.run

import pize.Pize.create
import pize.Pize.run
import pize.Pize.window
import pize.app.AppAdapter
import pize.graphics.gl.DepthFunc
import pize.graphics.gl.Gl.clearColorDepthBuffers
import pize.graphics.gl.Gl.depthFunc
import pize.graphics.gl.Gl.enable
import pize.graphics.gl.Target
import pize.tests.minecraft.game.Session
import pize.tests.minecraft.utils.log.Logger

class Minecraft : AppAdapter() {
    private var session: Session? = null
    override fun init() {
        enable(Target.DEPTH_TEST)
        depthFunc(DepthFunc.LEQUAL)
        Thread.currentThread().setName("Render Thread")
        session = Session()
    }

    override fun render() {
        clearColorDepthBuffers()
        session!!.render()
    }

    override fun resize(width: Int, height: Int) {
        session!!.resize(width, height)
    }

    override fun dispose() {
        session!!.dispose()
        Logger.Companion.instance().info("EXIT.")
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            create("Minecraft", 925, 640)
            window()!!.setIcon("icon.png")
            run(Minecraft())
        }
    }
}
